tests/matchers_unittest.py

```diff
@@ -0,0 +1,259 @@
+
+
+class StringTest(testslide.TestCase):
+    def testAnyStr(self):
+        self.assertEqual(testslide.matchers.AnyStr(), "aa")
+        self.assertEqual(testslide.matchers.AnyStr(), "")
+        self.assertNotEqual(testslide.matchers.AnyStr(), 69)
+        self.assertNotEqual(testslide.matchers.AnyStr(), None)
+
+    def testRegexMatches(self):
+        self.assertEqual(testslide.matchers.RegexMatches("b[aeiou]t"), "bott")
+        self.assertNotEqual(testslide.matchers.RegexMatches("b[aeiou]t"), "boot")
+        self.assertNotEqual(testslide.matchers.RegexMatches("b[aeiou]t"), 13)
+
+
+class TestSubsets(testslide.TestCase):
+    def testListContaining(self):
+        self.assertEqual(testslide.matchers.ListContaining([1, 2]), [1, 2, 3])
+        self.assertNotEqual(testslide.matchers.ListContaining([1, 2]), [2, 3, 4])
+        self.assertNotEqual(testslide.matchers.ListContaining([1, 2]), "DERP")
+
+    def testDictContaining(self):
+        self.assertEqual(
+            testslide.matchers.DictContaining({"a": "b", "c": 1}),
+            {"a": "b", "c": 1, "d": "e"},
+        )
+        self.assertNotEqual(
+            testslide.matchers.DictContaining({"a": "b", "c": 1}), {"c": 1, "d": "e"}
+        )
+        self.assertNotEqual(testslide.matchers.DictContaining([1, 2]), "DERP")
+
+
+class TestChaining(testslide.TestCase):
+    def testBitwiseAnd(self):
+        self.assertTrue(
+            isinstance(
+                testslide.matchers.Any() & testslide.matchers.AnyStr(),
+                testslide.matchers._AndMatcher,
+            )
+        )
+        self.assertEqual(
+            testslide.matchers.Any() & testslide.matchers.AnyStr(), "a"
+        )
+        self.assertNotEqual(
+            testslide.matchers.Any() & testslide.matchers.AnyStr(), 3
+        )
+
+    def testBitwiseOr(self):
+        self.assertTrue(
+            isinstance(
+                testslide.matchers.Any() | testslide.matchers.AnyStr(),
+                testslide.matchers._OrMatcher,
+            )
+        )
+        self.assertEqual(
+            testslide.matchers.AnyInt() | testslide.matchers.AnyStr(), "a"
+        )
+        self.assertEqual(
+            testslide.matchers.AnyInt() | testslide.matchers.AnyStr(), 3
+        )
+        self.assertNotEqual(
+            testslide.matchers.AnyInt() | testslide.matchers.AnyStr(), []
+        )
+
+    def testBitwiseXor(self):
+        self.assertTrue(
+            isinstance(
+                testslide.matchers.Any() ^ testslide.matchers.AnyStr(),
+                testslide.matchers._XorMatcher,
+            )
+        )
+        self.assertEqual(
+            testslide.matchers.AnyInt() ^ testslide.matchers.AnyStr(), []
+        )
+        self.assertNotEqual(
+            testslide.matchers.AnyInt() ^ testslide.matchers.AnyStr(), 3
+        )
+
+    def testCannotChainMoreThanTwo(self):
+        with self.assertRaises(testslide.matchers.AlreadyChainedException):
```

Comment:
Better use assertRaisesRegex(), as we don't have the possibility of the test passing with some other exception (eg: typo and a NameError).